### ISSUE TITLE
Can we support bfloat16 and float16 SIMD

### ISSUE THREAD

**Issue by chasingegg** [gorse-io/goat#12]
Although bfloat16 and float16 doest not natively supported by go and C++, but in pure C++ we could use SIMD, can we do this in go-cpp translation?

**Comment by zhenghaoz** [comment]
Yes, it supports bfloat16 and float16. We write a post recently:
[BF16 in the Go Programming Language](https://gorse.io/posts/bf16-in-golang.html)
